compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -130,6 +130,10 @@ public static ConstantValueResolver getInstance(CompilerContext context) {
         return constantValueResolver;
     }
 
+    public void setCurrentPackageId(PackageID packageID) {
```

Comment:
Instead of doing this can we see if we can enforce passing the package ID where resolve methods are called? That way we can ensure it works wherever it is called.

Reply:
I also thought that I approach first. But the null pointer occurring place is getting invoked from [here](https://github.com/ballerina-platform/ballerina-lang/blob/7d6951bf42d40fb80482dabc30b58fb8739c9abb/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java#L874).
In this place we are passing the class level variable.

Reply:
We can have that method (overloaded probably) also accept package ID?

Reply:
Updated with [commit](https://github.com/ballerina-platform/ballerina-lang/pull/43163/commits/0da8682df69d8d5409897a2a6ce40c998f1a523c).